test/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsServiceSpec.scala

```diff
@@ -0,0 +1,129 @@
+
+      hIS.getRepoToTeams.futureValue shouldBe Map.empty
+    }
+
+    "return Map.empty when TeamsAndReposConnector returns None repo for team" in new Setup {
+      when(mockTARC.teamsWithRepositories) thenReturn
+        Future.successful(Seq(Team(TeamName("foo"), None, None, None, repos = None)))
+
+      hIS.getRepoToTeams.futureValue shouldBe Map.empty
+    }
+
+    "return Map.empty when TeamsAndReposConnector returns empty repoType map for a team" in new Setup {
+      when(mockTARC.teamsWithRepositories) thenReturn
+        Future.successful(Seq(Team(TeamName("foo"), None, None, None, repos = Some(Map.empty))))
+
+      hIS.getRepoToTeams.futureValue shouldBe Map.empty
+    }
+
+    "return all teams for one repo when TeamsAndReposConnector returns multiple teams with only one repo" in new Setup {
+      when(mockTARC.teamsWithRepositories) thenReturn
+        Future.successful(
+          Seq(
+            Team(TeamName("foo"), None, None, None, Some(Map("Service"   -> Seq("auth")))),
+            Team(TeamName("bar"), None, None, None, Some(Map("Prototype" -> Seq("auth")))),
+            Team(TeamName("hello"), None, None, None, Some(Map("Other"   -> Seq("auth"))))
+          ))
+
+      hIS.getRepoToTeams.futureValue shouldBe Map("auth" -> Set("foo", "bar", "hello"))
+    }
+
+    "return one team for many repos when TeamsAndRepositoriesConnector returns one team with multiple repos" in new Setup {
+      when(mockTARC.teamsWithRepositories) thenReturn
+        Future.successful(
+          Seq(
+            Team(TeamName("foo"), None, None, None, Some(Map("Service" -> Seq("auth"), "Prototype" -> Seq("admin"), "Other" -> Seq("advert"))))
+          ))
+
+      hIS.getRepoToTeams.futureValue shouldBe Map("auth" -> Set("foo"), "admin" -> Set("foo"), "advert" -> Set("foo"))
+    }
+
+    "return correct map when TeamsAndRepositories returns multiple teams with some duplicate repositories" in new Setup {
+      when(mockTARC.teamsWithRepositories) thenReturn
+        Future.successful(
+          Seq(
+            Team(TeamName("foo"), None, None, None, Some(Map("Service" -> Seq("auth"), "Prototype"  -> Seq("admin"), "Other" -> Seq("advert", "world")))),
+            Team(TeamName("bar"), None, None, None, Some(Map("Service" -> Seq("hello"), "Prototype" -> Seq("jfrog"), "Other" -> Seq("advert"), "Stub" -> Seq("get"))))
+          ))
+
+      hIS.getRepoToTeams.futureValue shouldBe
+        Map("auth" -> Set("foo"), "admin" -> Set("foo"), "advert" -> Set("foo", "bar"), "world" -> Set("foo"), "hello" -> Set("bar"), "jfrog" -> Set("bar"), "get" -> Set("bar"))
+
+    }
+  }
+  private[this] trait Setup {
+    implicit val hc: HeaderCarrier              = HeaderCarrier()
+    val mockTARC: TeamsAndRepositoriesConnector = mock[TeamsAndRepositoriesConnector]
+    val mockHIC: HealthIndicatorsConnector      = mock[HealthIndicatorsConnector]
+    val hIS                                     = new HealthIndicatorsService(mockTARC, mockHIC)
```

Comment:
I appreciate what you've done here, but I would prefer the names to be spelled out explicitly.
TARC, HIC, and hIS are not common acronyms, and I have to mentally convert them every time I read them :(